src/wp-includes/block-template.php

```diff
@@ -224,37 +223,16 @@ function get_the_block_template_html() {
 		return;
 	}
 
-	$content = $wp_embed->run_shortcode( $_wp_current_template_content );
-	$content = $wp_embed->autoembed( $content );
-	$content = shortcode_unautop( $content );
-	$content = do_shortcode( $content );
+	$content = $_wp_current_template_content;
 
-	/*
-	 * Most block themes omit the `core/query` and `core/post-template` blocks in their singular content templates.
-	 * While this technically still works since singular content templates are always for only one post, it results in
-	 * the main query loop never being entered which causes bugs in core and the plugin ecosystem.
+	/**
+	 * Filters the block template HTML content.
 	 *
-	 * The workaround below ensures that the loop is started even for those singular templates. The while loop will by
-	 * definition only go through a single iteration, i.e. `do_blocks()` is only called once. Additional safeguard
-	 * checks are included to ensure the main query loop has not been tampered with and really only encompasses a
-	 * single post.
+	 * @since 6.4.0
 	 *
-	 * Even if the block template contained a `core/query` and `core/post-template` block referencing the main query
-	 * loop, it would not cause errors since it would use a cloned instance and go through the same loop of a single
-	 * post, within the actual main query loop.
+	 * @param string $content The entire block template HTML content.
 	 */
-	if ( is_singular() && 1 === $wp_query->post_count && have_posts() ) {
-		while ( have_posts() ) {
-			the_post();
-			$content = do_blocks( $content );
-		}
-	} else {
-		$content = do_blocks( $content );
-	}
-
-	$content = wptexturize( $content );
-	$content = convert_smilies( $content );
-	$content = wp_filter_content_tags( $content, 'template' );
+	$content = apply_filters( 'template', $content );
```

Comment:
This filter name is already in use in `get_template`.

Reply:
This is really power filter, are we sure we want to let developers filter the all the markup of the page?

Reply:
I think that's okay. It's similar to `the_content`, and it allows customizing the callbacks. Of course it's all a matter of responsible usage, but I'd argue already today in core any filter can be misused if someone wants to.

If we feel this is too much, maybe we can implement some alternative. Maybe just a list of callback functions that can be filtered only to remove any. Or we could make it an action so that developers can add callbacks that "listen" to the template content but cannot modify it. WDYT?

Reply:
Such a power filter is what is requested in Core-43258, although for the entire page instead of just between `wp_body_open()` and `wp_footer()`, as I commented in https://github.com/WordPress/wordpress-develop/pull/5188#discussion_r1323592176. If we add yet another filter for the entire document in addition to this block template HTML filter, would that be problematic? Doesn't seem so.

Reply:
I agree there is overlap here, but I would prefer to not mix the output buffering discussion into this change here since it doesn't require it.

I am curious though, for block themes specifically, wouldn't this filter here already be sufficient? If we had this, I'm not sure about the benefits block themes would get from using an output buffer. Even without filtering the entire document, I can't think of any real limitations, since the filter would run before `wp_head`, `wp_body_open` and `wp_footer`, so you could still affect pretty much all other parts of the HTML document as well at that point.